src/transport_performance/population/rasterpop.py

```diff
@@ -0,0 +1,707 @@
+"""Class to handle raster population data."""
+
+import geopandas as gpd
+import os
+import numpy as np
+import rasterio as rio
+import rioxarray
+import folium
+import cartopy.crs as ccrs
+import cartopy.io.img_tiles as cimgt
+import matplotlib.pyplot as plt
+
+from datetime import datetime
+from geocube.vector import vectorize
+from typing import Union, Type, Tuple
+from shapely.geometry.polygon import Polygon
+from matplotlib import colormaps
+from cartopy.mpl.geoaxes import GeoAxes
+
+
+class RasterPop:
+    """Prepare raster population inputs for trasport analysis.
+
+    This class is suited to working with rastered population data (e.g.
+    gridded population estimates).
+
+    Parameters
+    ----------
+    filepath : Union[str, bytes, os.PathLike]
+        File path to population data
+
+    Attributes
+    ----------
+    pop_gdf : gpd.GeoDataFrame
+        A geopandas dataframe of the input data, with gridded geometry. This is
+        in the same CRS as the input raster data.
+    centroid_gdf
+        A geopandas dataframe of grid centroids, converted to EPSG:4326 for
+        transport analysis.
+
+    Methods
+    -------
+    get_pop
+        Read and preprocess population estimates into a geopandas dataframe.
+    plot
+        Build static and interactive visualisations of population data. Can
+        only use this method once `get_pop` has been called.
+
+    Raises
+    ------
+    FileNotFoundError
+        When `filepath` is not a file or can not be found.
+
+    """
+
+    def __init__(self, filepath: Union[str, bytes, os.PathLike]) -> None:
+
+        # defend against cases where input is not a file and does not exist
+        if not os.path.isfile(filepath):
+            raise FileNotFoundError(f"{filepath} is not a file.")
+
+        self.__filepath = filepath
+
+        # record the crs of the data source without reading in data
+        with rio.open(filepath) as src:
+            self.__crs = src.crs.to_string()
+
+        # set attributes to None to prevent plotting function calls
+        self.pop_gdf = None
+        self._uc_gdf = None
+
+    def get_pop(
+        self,
+        aoi_bounds: Type[Polygon],
```

Comment:
Better to have this read in the entire population file if no `aoi_bound` polygon is provided? I presume this is potentially poorly performant? Agreed to raise as feature request.